### PR TITLE
Initial work on nodes.json support and node classes.

### PR DESCRIPTION
**Pull request by kaendfinger**

### CODE REVIEW

**Comment by logangorence** [comment]
Approved

File: DSLink/DSLinkContainer.cs [hunk 1]
```diff
@@ -61,11 +61,42 @@ public DSLinkContainer(Configuration config) : base(config)
             _pingTask = Task.Factory.StartNew(OnPingElapsed);
         }
 
+        private bool _isInitialized = false;
```
Comment (logangorence) [inline DSLink/DSLinkContainer.cs]: Move this above the constructor.

**Comment by kaendfinger** [comment]
PR is not ready for review.

File: DSLink/Nodes/Node.cs [hunk 2]
```diff
@@ -217,6 +225,24 @@ public Node(string name, Node parent, AbstractContainer link)
             }
         }
 
+        private bool _initializedClass = false;
```
Comment (logangorence) [inline DSLink/Nodes/Node.cs]: Move above constructor.

File: DSLink/Respond/Responder.cs [hunk 3]
```diff
@@ -44,6 +49,81 @@ internal Responder(AbstractContainer link)
...
+
+            if (file != null)
+            {
+                string data = obj.ToString();
+                await file.WriteAllTextAsync(data);
+                string path = file.Path;
+                if (_link.Config.LogLevel.DoesPrint(LogLevel.Debug))
+                {
+                    _link.Logger.Debug($"Wrote {data} to {path}");
+
```
Comment (logangorence) [inline DSLink/Respond/Responder.cs]: Remove blank line

File: DSLink/Respond/Responder.cs [hunk 4]
```diff
@@ -34,6 +37,8 @@ public sealed class Responder
         /// </summary>
         internal StreamManager StreamManager;
 
+        internal IDictionary<string, Action<Node>> NodeClasses;
```
Comment (logangorence) [inline DSLink/Respond/Responder.cs]: Document

File: RNG/RNG.cs [hunk 5]
```diff
@@ -14,62 +14,82 @@ public class ExampleDSLink : DSLinkContainer
...
                 {
-                    value.Set(num++);
-                }
+                node.Writable = Permission.Read;
+                node.ValueType = ValueType.Number;
+                node.Value.Set(0.1);
+                rngs.Add(node.Value);
             });
+
+            Task.Run(async () =>
```
Comment (logangorence) [inline RNG/RNG.cs]: Fix tabbing

**Comment by logangorence** [comment]
Approved

**Pull request merged**